### ISSUE TITLE
Compiling my own copy?

### ISSUE THREAD

**Issue by mprowe** [chickey/RS485-WiFi-EPEver#14]
I see the pre-compiled files and I see the src/RS485-WiFi-EPEver.ino.
How can I compile the .ino in the Arduino IDE? I'd like to make one or two tweaks...

Regards, M

**Comment by chickey** [comment]
Hi Martin, all the code is there in the ino and associated files, once you've got the dependencies installed you can compile.  If you use vscode it ensures that the relevant libraries are installed.

**Comment by mprowe** [comment]
Hi Colin,
_"once you've got the dependencies installed you can compile."_ 
That is my problem. Do you have a list of those dependencies? Or can I ask which is the library that will resolve this error?

`/home/mprowe/Arduino/libraries/ESPUI/src/ESPUI.h:23:10: fatal error: ESPAsyncTCP.h: No such file or directory
23 | #include <ESPAsyncTCP.h>
|                ^~~~~~~~~~~~~~~
compilation terminated.`

I have tried the similar sounding ESPAsync_WiFiManager" & ESPAsync_WiFiManager_Lite both by Khoi Hoang from the Arduino Library Manager with no luck.

And forgive my ignorance, but I do not understand what "_If you use vscode it ensures that the relevant libraries are installed."_ means. Is this the sentence that unlocks the way forward?
Regards, M.

**Comment by artlab23** [comment]
This may be the same _ESP8266HTTPClient_ issue as was tripping me up a year ago (we [discussed](https://discord.com/channels/605714448002449457/810862767187689484/853397911576772618) on the Discord channel).

```
/*
 * Since both begin() functions take a reference to client as a parameter, you need to 
 * ensure the client object lives the entire time of the HTTPClient
 */
    bool begin(WiFiClient &client, const String& url);
    bool begin(WiFiClient &client, const String& host, uint16_t port, const String& uri = "/", bool https = false);

    // old API is now explicitly forbidden
```

What I did to get it (the 0.60 code) to compile:

Moved line 85 of influxdb.h:

`  WiFiClient client;`

..up to line 75.

Changed line 76 from:

`  http.begin(url);`

..to:

`  http.begin(client, url);`

This compiles OK* in Arduino IDE for me across 3 different (fairly well updated) Linux machines, uploads and runs fine (webUI + feeding influxdb, I don't use MQTT).

(*A bit of whining about SPIFFS being deprecated).

This all only came up today as I borked mine with an OTA update because I fancied having the 'load' switch at the top of the first tab...  It had been sitting out there chugging away for a whole year without demanding attention.  So thanks again, Col, for all your work on this.  Much appreciated. :)
